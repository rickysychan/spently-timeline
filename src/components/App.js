import React, {Component} from 'react'
import Data from '../data/data.json'
import Table from './Table'
import { sortByLowest } from '../helper'

export default class App extends Component{
    state = {
        data: []
    }
    componentWillMount(){
        const sortedData = sortByLowest(Data, 'timestamp')
        this.setState({ data: sortedData })
    }
    
    render(){
        return(
            <div className='table-timeline'>
                { this.state.data.map( client => {
                        return(
                            <Table 
                                key={client._id}
                                email={client.email} 
                                emailType={client.emailType}
                                timestamp={parseInt((client.timestamp/1000).toString(), 10)}
                            />
                        )
                    })
                }
            </div>
        )
    }
}
