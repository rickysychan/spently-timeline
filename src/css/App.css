.table{
    width: 100%;
    height: 300px;
    border-color: var(--border-color);
    background-color: white;
    white-space: nowrap;
    overflow: scroll;
    border-top: 2px solid var(--light-gray);
}
.card{
    border-spacing: 0;
    border: 1px solid;
    border-color: var(--border-color);
    background-color: white;
    border-radius: 5px;    
}
table{
    width: 100%;
}
#table-header{
    font-size: 20px; 
}
.subheading-left{
    text-align: left;
    padding-left: 5px;
}
.subheading-right{
    text-align: right;
    padding-right: 5px;
}
.subheading-email:hover{
    background-color: var(--teal-light)
}
.subheading-emailType:hover{
    background-color: var(--teal-light)
}
.subheading-timestamp:hover{
    background-color: var(--teal-light)
}