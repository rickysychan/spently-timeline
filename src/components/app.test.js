import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import data from '../data/data.json'

const app = shallow(<App />)

describe('app', () => {
    it('should render properly', () => {
        expect(app).toMatchSnapshot()
    })    
    describe('when the component initializes', () => {
        it('should retrieve the data and set it in `state`', () => {
            expect(app.state().data).toEqual(data)
        })
    })
    it('renders a list of emails from data', () => {
        expect(app.find('.table-timeline').children().length).toEqual(data.length)
    })
})