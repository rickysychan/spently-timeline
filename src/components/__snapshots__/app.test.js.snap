// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Table
          email="marcie.keller@puria.co.uk"
          emailType="Newsletter"
          timestamp={1534791896}
/>,
        <Table
          email="sharp.sanchez@exiand.org"
          emailType="Newsletter"
          timestamp={1530837537}
/>,
        <Table
          email="coffey.bishop@poshome.tv"
          emailType="POS Receipt"
          timestamp={1529280627}
/>,
        <Table
          email="oliver.lewis@kozgene.net"
          emailType="Shipping Update"
          timestamp={1526556220}
/>,
        <Table
          email="bonner.roth@hometown.ca"
          emailType="Customer Password Reset"
          timestamp={1519381111}
/>,
        <Table
          email="miriam.vasquez@playce.com"
          emailType="Abandoned Checkout"
          timestamp={1518621000}
/>,
        <Table
          email="bettie.washington@eplosion.biz"
          emailType="Order Confirmation"
          timestamp={1515214989}
/>,
        <Table
          email="lenora.marshall@furnitech.info"
          emailType="Shipment Delivered"
          timestamp={1512952979}
/>,
        <Table
          email="hester.rich@cemention.org"
          emailType="Draft Invoice"
          timestamp={1511440881}
/>,
        <Table
          email="martina.salazar@cyclonica.tv"
          emailType="Shipment Delivered"
          timestamp={1510524050}
/>,
        <Table
          email="webster.blackburn@dentrex.com"
          emailType="Order Cancelled"
          timestamp={1504189812}
/>,
        <Table
          email="campos.terry@tripsch.biz"
          emailType="POS Receipt"
          timestamp={1503391809}
/>,
        <Table
          email="whitney.joyce@zillidium.co.uk"
          emailType="Shipment Sent"
          timestamp={1493010976}
/>,
        <Table
          email="kristy.bell@crustatia.org"
          emailType="Gift Card"
          timestamp={1490717952}
/>,
        <Table
          email="virgie.chandler@nurali.biz"
          emailType="Contact Customer"
          timestamp={1480955338}
/>,
        <Table
          email="myrtle.montgomery@emtrak.me"
          emailType="Newsletter"
          timestamp={1475241319}
/>,
        <Table
          email="jannie.house@bedder.us"
          emailType="Shipment Sent"
          timestamp={1470911209}
/>,
        <Table
          email="regina.rocha@interfind.me"
          emailType="Contact Customer"
          timestamp={1466948304}
/>,
        <Table
          email="camille.farmer@terragen.us"
          emailType="Shipping Confirmation"
          timestamp={1455788435}
/>,
        <Table
          email="beasley.frazier@furnafix.info"
          emailType="Refund Notification"
          timestamp={1452582699}
/>,
        <Table
          email="cassandra.neal@pyramax.name"
          emailType="Shipment Sent"
          timestamp={1436589207}
/>,
        <Table
          email="lucia.jensen@netur.biz"
          emailType="Newsletter"
          timestamp={1434489029}
/>,
        <Table
          email="denise.williams@zoinage.us"
          emailType="Refund Notification"
          timestamp={1427677678}
/>,
        <Table
          email="ursula.estes@zolarex.name"
          emailType="Newsletter"
          timestamp={1424134270}
/>,
        <Table
          email="reynolds.moran@zytrac.ca"
          emailType="Shipping Update"
          timestamp={1403774908}
/>,
        <Table
          email="anastasia.durham@ginkogene.net"
          emailType="Order Cancelled"
          timestamp={1394601773}
/>,
      ],
      "className": "table-timeline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "5b7ddb2171df5789191554bb",
        "nodeType": "function",
        "props": Object {
          "email": "marcie.keller@puria.co.uk",
          "emailType": "Newsletter",
          "timestamp": 1534791896,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb212403c9cee7a0764f",
        "nodeType": "function",
        "props": Object {
          "email": "sharp.sanchez@exiand.org",
          "emailType": "Newsletter",
          "timestamp": 1530837537,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21f838e0a5076ae5a7",
        "nodeType": "function",
        "props": Object {
          "email": "coffey.bishop@poshome.tv",
          "emailType": "POS Receipt",
          "timestamp": 1529280627,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb211a2867ff74585065",
        "nodeType": "function",
        "props": Object {
          "email": "oliver.lewis@kozgene.net",
          "emailType": "Shipping Update",
          "timestamp": 1526556220,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21b6cf1d7b2b78ba88",
        "nodeType": "function",
        "props": Object {
          "email": "bonner.roth@hometown.ca",
          "emailType": "Customer Password Reset",
          "timestamp": 1519381111,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb216d1fcb619dbe8145",
        "nodeType": "function",
        "props": Object {
          "email": "miriam.vasquez@playce.com",
          "emailType": "Abandoned Checkout",
          "timestamp": 1518621000,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb213530115c987aefcc",
        "nodeType": "function",
        "props": Object {
          "email": "bettie.washington@eplosion.biz",
          "emailType": "Order Confirmation",
          "timestamp": 1515214989,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb2190c6d34ad86bb36c",
        "nodeType": "function",
        "props": Object {
          "email": "lenora.marshall@furnitech.info",
          "emailType": "Shipment Delivered",
          "timestamp": 1512952979,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb217e7a0bdcf6e8545c",
        "nodeType": "function",
        "props": Object {
          "email": "hester.rich@cemention.org",
          "emailType": "Draft Invoice",
          "timestamp": 1511440881,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21ea9816bc0195aa18",
        "nodeType": "function",
        "props": Object {
          "email": "martina.salazar@cyclonica.tv",
          "emailType": "Shipment Delivered",
          "timestamp": 1510524050,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21540230519f464c96",
        "nodeType": "function",
        "props": Object {
          "email": "webster.blackburn@dentrex.com",
          "emailType": "Order Cancelled",
          "timestamp": 1504189812,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb2135e3973fabb1cf41",
        "nodeType": "function",
        "props": Object {
          "email": "campos.terry@tripsch.biz",
          "emailType": "POS Receipt",
          "timestamp": 1503391809,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb214bd32e72f5666a9b",
        "nodeType": "function",
        "props": Object {
          "email": "whitney.joyce@zillidium.co.uk",
          "emailType": "Shipment Sent",
          "timestamp": 1493010976,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb2152b4f95afa96b211",
        "nodeType": "function",
        "props": Object {
          "email": "kristy.bell@crustatia.org",
          "emailType": "Gift Card",
          "timestamp": 1490717952,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21333e572d2d44431c",
        "nodeType": "function",
        "props": Object {
          "email": "virgie.chandler@nurali.biz",
          "emailType": "Contact Customer",
          "timestamp": 1480955338,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21ee79d9942947c9c4",
        "nodeType": "function",
        "props": Object {
          "email": "myrtle.montgomery@emtrak.me",
          "emailType": "Newsletter",
          "timestamp": 1475241319,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb219e3a12f2cb00662d",
        "nodeType": "function",
        "props": Object {
          "email": "jannie.house@bedder.us",
          "emailType": "Shipment Sent",
          "timestamp": 1470911209,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21829fb0e15bed5da0",
        "nodeType": "function",
        "props": Object {
          "email": "regina.rocha@interfind.me",
          "emailType": "Contact Customer",
          "timestamp": 1466948304,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21df85df05ae8dce24",
        "nodeType": "function",
        "props": Object {
          "email": "camille.farmer@terragen.us",
          "emailType": "Shipping Confirmation",
          "timestamp": 1455788435,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21d983142aa62016f5",
        "nodeType": "function",
        "props": Object {
          "email": "beasley.frazier@furnafix.info",
          "emailType": "Refund Notification",
          "timestamp": 1452582699,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21734bf5bab7250352",
        "nodeType": "function",
        "props": Object {
          "email": "cassandra.neal@pyramax.name",
          "emailType": "Shipment Sent",
          "timestamp": 1436589207,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb2145d1c16d45ebcea8",
        "nodeType": "function",
        "props": Object {
          "email": "lucia.jensen@netur.biz",
          "emailType": "Newsletter",
          "timestamp": 1434489029,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21a5ce9636dc1d7e41",
        "nodeType": "function",
        "props": Object {
          "email": "denise.williams@zoinage.us",
          "emailType": "Refund Notification",
          "timestamp": 1427677678,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb21caae970a5fbc9fb6",
        "nodeType": "function",
        "props": Object {
          "email": "ursula.estes@zolarex.name",
          "emailType": "Newsletter",
          "timestamp": 1424134270,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb219559519edfaf281d",
        "nodeType": "function",
        "props": Object {
          "email": "reynolds.moran@zytrac.ca",
          "emailType": "Shipping Update",
          "timestamp": 1403774908,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5b7ddb2100931b8e4b578a1b",
        "nodeType": "function",
        "props": Object {
          "email": "anastasia.durham@ginkogene.net",
          "emailType": "Order Cancelled",
          "timestamp": 1394601773,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Table
            email="marcie.keller@puria.co.uk"
            emailType="Newsletter"
            timestamp={1534791896}
/>,
          <Table
            email="sharp.sanchez@exiand.org"
            emailType="Newsletter"
            timestamp={1530837537}
/>,
          <Table
            email="coffey.bishop@poshome.tv"
            emailType="POS Receipt"
            timestamp={1529280627}
/>,
          <Table
            email="oliver.lewis@kozgene.net"
            emailType="Shipping Update"
            timestamp={1526556220}
/>,
          <Table
            email="bonner.roth@hometown.ca"
            emailType="Customer Password Reset"
            timestamp={1519381111}
/>,
          <Table
            email="miriam.vasquez@playce.com"
            emailType="Abandoned Checkout"
            timestamp={1518621000}
/>,
          <Table
            email="bettie.washington@eplosion.biz"
            emailType="Order Confirmation"
            timestamp={1515214989}
/>,
          <Table
            email="lenora.marshall@furnitech.info"
            emailType="Shipment Delivered"
            timestamp={1512952979}
/>,
          <Table
            email="hester.rich@cemention.org"
            emailType="Draft Invoice"
            timestamp={1511440881}
/>,
          <Table
            email="martina.salazar@cyclonica.tv"
            emailType="Shipment Delivered"
            timestamp={1510524050}
/>,
          <Table
            email="webster.blackburn@dentrex.com"
            emailType="Order Cancelled"
            timestamp={1504189812}
/>,
          <Table
            email="campos.terry@tripsch.biz"
            emailType="POS Receipt"
            timestamp={1503391809}
/>,
          <Table
            email="whitney.joyce@zillidium.co.uk"
            emailType="Shipment Sent"
            timestamp={1493010976}
/>,
          <Table
            email="kristy.bell@crustatia.org"
            emailType="Gift Card"
            timestamp={1490717952}
/>,
          <Table
            email="virgie.chandler@nurali.biz"
            emailType="Contact Customer"
            timestamp={1480955338}
/>,
          <Table
            email="myrtle.montgomery@emtrak.me"
            emailType="Newsletter"
            timestamp={1475241319}
/>,
          <Table
            email="jannie.house@bedder.us"
            emailType="Shipment Sent"
            timestamp={1470911209}
/>,
          <Table
            email="regina.rocha@interfind.me"
            emailType="Contact Customer"
            timestamp={1466948304}
/>,
          <Table
            email="camille.farmer@terragen.us"
            emailType="Shipping Confirmation"
            timestamp={1455788435}
/>,
          <Table
            email="beasley.frazier@furnafix.info"
            emailType="Refund Notification"
            timestamp={1452582699}
/>,
          <Table
            email="cassandra.neal@pyramax.name"
            emailType="Shipment Sent"
            timestamp={1436589207}
/>,
          <Table
            email="lucia.jensen@netur.biz"
            emailType="Newsletter"
            timestamp={1434489029}
/>,
          <Table
            email="denise.williams@zoinage.us"
            emailType="Refund Notification"
            timestamp={1427677678}
/>,
          <Table
            email="ursula.estes@zolarex.name"
            emailType="Newsletter"
            timestamp={1424134270}
/>,
          <Table
            email="reynolds.moran@zytrac.ca"
            emailType="Shipping Update"
            timestamp={1403774908}
/>,
          <Table
            email="anastasia.durham@ginkogene.net"
            emailType="Order Cancelled"
            timestamp={1394601773}
/>,
        ],
        "className": "table-timeline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "5b7ddb2171df5789191554bb",
          "nodeType": "function",
          "props": Object {
            "email": "marcie.keller@puria.co.uk",
            "emailType": "Newsletter",
            "timestamp": 1534791896,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb212403c9cee7a0764f",
          "nodeType": "function",
          "props": Object {
            "email": "sharp.sanchez@exiand.org",
            "emailType": "Newsletter",
            "timestamp": 1530837537,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21f838e0a5076ae5a7",
          "nodeType": "function",
          "props": Object {
            "email": "coffey.bishop@poshome.tv",
            "emailType": "POS Receipt",
            "timestamp": 1529280627,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb211a2867ff74585065",
          "nodeType": "function",
          "props": Object {
            "email": "oliver.lewis@kozgene.net",
            "emailType": "Shipping Update",
            "timestamp": 1526556220,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21b6cf1d7b2b78ba88",
          "nodeType": "function",
          "props": Object {
            "email": "bonner.roth@hometown.ca",
            "emailType": "Customer Password Reset",
            "timestamp": 1519381111,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb216d1fcb619dbe8145",
          "nodeType": "function",
          "props": Object {
            "email": "miriam.vasquez@playce.com",
            "emailType": "Abandoned Checkout",
            "timestamp": 1518621000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb213530115c987aefcc",
          "nodeType": "function",
          "props": Object {
            "email": "bettie.washington@eplosion.biz",
            "emailType": "Order Confirmation",
            "timestamp": 1515214989,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb2190c6d34ad86bb36c",
          "nodeType": "function",
          "props": Object {
            "email": "lenora.marshall@furnitech.info",
            "emailType": "Shipment Delivered",
            "timestamp": 1512952979,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb217e7a0bdcf6e8545c",
          "nodeType": "function",
          "props": Object {
            "email": "hester.rich@cemention.org",
            "emailType": "Draft Invoice",
            "timestamp": 1511440881,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21ea9816bc0195aa18",
          "nodeType": "function",
          "props": Object {
            "email": "martina.salazar@cyclonica.tv",
            "emailType": "Shipment Delivered",
            "timestamp": 1510524050,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21540230519f464c96",
          "nodeType": "function",
          "props": Object {
            "email": "webster.blackburn@dentrex.com",
            "emailType": "Order Cancelled",
            "timestamp": 1504189812,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb2135e3973fabb1cf41",
          "nodeType": "function",
          "props": Object {
            "email": "campos.terry@tripsch.biz",
            "emailType": "POS Receipt",
            "timestamp": 1503391809,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb214bd32e72f5666a9b",
          "nodeType": "function",
          "props": Object {
            "email": "whitney.joyce@zillidium.co.uk",
            "emailType": "Shipment Sent",
            "timestamp": 1493010976,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb2152b4f95afa96b211",
          "nodeType": "function",
          "props": Object {
            "email": "kristy.bell@crustatia.org",
            "emailType": "Gift Card",
            "timestamp": 1490717952,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21333e572d2d44431c",
          "nodeType": "function",
          "props": Object {
            "email": "virgie.chandler@nurali.biz",
            "emailType": "Contact Customer",
            "timestamp": 1480955338,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21ee79d9942947c9c4",
          "nodeType": "function",
          "props": Object {
            "email": "myrtle.montgomery@emtrak.me",
            "emailType": "Newsletter",
            "timestamp": 1475241319,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb219e3a12f2cb00662d",
          "nodeType": "function",
          "props": Object {
            "email": "jannie.house@bedder.us",
            "emailType": "Shipment Sent",
            "timestamp": 1470911209,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21829fb0e15bed5da0",
          "nodeType": "function",
          "props": Object {
            "email": "regina.rocha@interfind.me",
            "emailType": "Contact Customer",
            "timestamp": 1466948304,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21df85df05ae8dce24",
          "nodeType": "function",
          "props": Object {
            "email": "camille.farmer@terragen.us",
            "emailType": "Shipping Confirmation",
            "timestamp": 1455788435,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21d983142aa62016f5",
          "nodeType": "function",
          "props": Object {
            "email": "beasley.frazier@furnafix.info",
            "emailType": "Refund Notification",
            "timestamp": 1452582699,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21734bf5bab7250352",
          "nodeType": "function",
          "props": Object {
            "email": "cassandra.neal@pyramax.name",
            "emailType": "Shipment Sent",
            "timestamp": 1436589207,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb2145d1c16d45ebcea8",
          "nodeType": "function",
          "props": Object {
            "email": "lucia.jensen@netur.biz",
            "emailType": "Newsletter",
            "timestamp": 1434489029,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21a5ce9636dc1d7e41",
          "nodeType": "function",
          "props": Object {
            "email": "denise.williams@zoinage.us",
            "emailType": "Refund Notification",
            "timestamp": 1427677678,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb21caae970a5fbc9fb6",
          "nodeType": "function",
          "props": Object {
            "email": "ursula.estes@zolarex.name",
            "emailType": "Newsletter",
            "timestamp": 1424134270,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb219559519edfaf281d",
          "nodeType": "function",
          "props": Object {
            "email": "reynolds.moran@zytrac.ca",
            "emailType": "Shipping Update",
            "timestamp": 1403774908,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5b7ddb2100931b8e4b578a1b",
          "nodeType": "function",
          "props": Object {
            "email": "anastasia.durham@ginkogene.net",
            "emailType": "Order Cancelled",
            "timestamp": 1394601773,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
